src/api/consumer/lib/add.js

```diff
@@ -201,14 +167,20 @@ export default (async function addAction(
       ];
 
       // mark or add test files according to dsl
-      const newFileArrWithTests = await updateTestFilesAccordingToDsl(files, tests);
+      const testFilesArr = !R.isEmpty(tests) ? await getFiles(files.map(file => file.relativePath), tests) : [];
```

Comment:
This is the exact same lines as lines 127-133 you should extract it to a function